### PR TITLE
Create and check new alias

### PR DESCRIPTION
**Pull request by helltepakcer**
Removed all prints and sorted code.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,8 @@
 dist: trusty
+sudo: required
```
Comment (timbortnik) [inline .travis.yml]: It would be better to merge master to your branch. Copypasting may lead to merge conflicts

File: .travis.yml [hunk 2]
```diff
@@ -13,4 +17,4 @@ install:
   - pip install -r requirements.txt
 script:
   - python --version
-  - behave
+  - behave
```
Comment (timbortnik) [inline .travis.yml]: Any reason to remove end of line here?

File: features/environment_secret.py.sample [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # -*- coding: UTF-8 -*-
 import os
 
-HIPCHAT_LOGIN = os.environ.get("HCLOGIN", "me+test@gmail.com")
-HIPCHAT_PASS = os.environ.get("HCPASS", "testuserpassword")
+HIPCHAT_LOGIN = os.environ.get("HCLOGIN", "test+genasmetchik@gmail.com")
+HIPCHAT_PASS = os.environ.get("HCPASS", "Ry990sQ1")
```
Comment (timbortnik) [inline features/environment_secret.py.sample]: Any password exposed to open repository should be changed ASAP

File: features/steps/step_selenium_alias_hipchat.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+def step_impl(context):
+
+    context.authorized_page.enter_web_chat()
+    context.chat_page.open_chat()
+    context.chat_page.open_alias_menu()
+    context.chat_page.open_menu()
+    context.chat_page.open_config()
+    context.chat_page.input_data_in_alias_form()
+    context.chat_page.input_data_in_alias_name_form()
+    pass
```
Comment (timbortnik) [inline features/steps/step_selenium_alias_hipchat.py]: 'pass' looks redundant here

File: features/steps/step_selenium_alias_hipchat.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+# ----------------------------------------------------------------------------
+# STEPS:
+# ----------------------------------------------------------------------------
+from behave import given, when, then
+
+
+
+@given('we are on login page')
+def step_impl(context):
+    context.login_page.navigate()
```
Comment (timbortnik) [inline features/steps/step_selenium_alias_hipchat.py]: Do we have same code elsewhere? https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

File: pages/chat_page.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    def open_alias_menu(self):
+        self.find_plus_butt().click()
+
+    def find_plus_butt(self):
+        self.context.wait.until(EC.presence_of_element_located((By.ID, 'input_actions_dropdown-trigger')))
+        return self.context.driver.find_element_by_id("input_actions_dropdown-trigger")
+
+    def open_menu(self):
+        self.find_butt_in_dropdown_menu().click()
+
```
Comment (timbortnik) [inline pages/chat_page.py]: please "inspect code" in pycharm and follow PEP8 style (empty lines between code blocks in this case)

File: pages/chat_page.py [hunk 7]
```diff
@@ -0,0 +1,81 @@
+from .base_page import Page
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+
+class ChatPage(Page):
+
+    url = 'chat/'
+
+    def open_chat(self):
+        self.find_chat().click()
```
Comment (timbortnik) [inline pages/chat_page.py]: shall we wait for chat to open here?

File: pages/chat_page.py [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    def find_plus_butt(self):
+        self.context.wait.until(EC.presence_of_element_located((By.ID, 'input_actions_dropdown-trigger')))
+        return self.context.driver.find_element_by_id("input_actions_dropdown-trigger")
+
+    def open_menu(self):
+        self.find_butt_in_dropdown_menu().click()
+
+
+
+    def find_butt_in_dropdown_menu(self):
```
Comment (timbortnik) [inline pages/chat_page.py]: find_button?

File: pages/chat_page.py [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+        return self.context.driver.find_element_by_name('alias')
+
+    def input_data_in_alias_name_form(self):
+
+        self.find_form_name().send_keys('HenaYamkoviy')
+        self.find_form_name().send_keys(u'\ue007')
+        self.find_form_name().send_keys(u'\ue007')
+
+    def find_form_name(self):
+        return self.context.driver.find_element(By.XPATH, '//*[@id="react-app"]/div/form/div/div[2]/div/div/div/input')
```
Comment (timbortnik) [inline pages/chat_page.py]: Can we look for a better anchor than react-app?

File: pages/chat_page.py [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+    def find_input_but(self):
```
Comment (timbortnik) [inline pages/chat_page.py]: input_button?

**Review by timbortnik — CHANGES_REQUESTED** [review]

File: pages/chat_page.py (hunk 7 above)
Comment (helltepakcer) [reply pages/chat_page.py]: No, you right, I changed name of method

File: pages/chat_page.py (hunk 6 above)
Comment (helltepakcer) [reply pages/chat_page.py]: Done

File: pages/chat_page.py (hunk 8 above)
Comment (helltepakcer) [reply pages/chat_page.py]: I had rename method to find_button..(etc)

File: pages/chat_page.py (hunk 9 above)
Comment (helltepakcer) [reply pages/chat_page.py]: Yes, I change condition of this search

File: pages/chat_page.py (hunk 10 above)
Comment (helltepakcer) [reply pages/chat_page.py]: I had rename method to find_input_button, thank you for review and comments!

File: pages/authorized_page.py [hunk 11]
```diff
@@ -15,3 +15,8 @@ def label_page_head(self):
 
     def get_page_head(self):
         return self.label_page_head().text
+
+    def enter_web_chat(self):
+        #self.context.wait.until(lambda driver: driver.find_element_by_id('status_dropdown'))
```
Comment (SergeyMoroz0703) [inline pages/authorized_page.py]: remove unnecessary comment

File: pages/chat_page.py [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+    def find_added_element(self):
+        self.context.wait.until(
+            lambda driver: self.context.driver.find_elements_by_css_selector('.aui>.aliases>.alias>.mentions'))
+        table = self.context.driver.find_elements_by_css_selector('.aui>.aliases>.alias>.mentions')
+
+        result = False
+        for element in table:
+            if element.text == "@HenaYamkoviy":
+                result = True
+        return result
```
Comment (SergeyMoroz0703) [inline pages/chat_page.py]: Why we cant chage this to ?
```
if element.text == "@Hena....
       return True
```
Anyway this is not mistake.
And maybe need to change variable "@HenaYamkoviy" to something indepentendtly variable? what if we will not have this specific user in another chat?

File: features/hipchat_alias_check.feature [hunk 13]
```diff
@@ -0,0 +1,11 @@
+
+Feature: Check Alias Functional
+
+Scenario: Login, Open Alias Room, Create Alias, Check
+
+    Given we are on Hipchat Login Page
+    When we enter login
+    And we enter password
+    When we create new Alias
+    Then we check our data
```
Comment (timbortnik) [inline features/hipchat_alias_check.feature]: Could you be more explicit with this please? What do we check?

File: features/steps/step_selenium_alias_hipchat.py [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+    context.lobby_page.random_click()
+    context.lobby_page.open_alias_room()
+    assert context.lobby_page.chat_adding_alias()
+
+@when('we create new alias with click buttons')
+def step_impl(context):
+    context.lobby_page.delete_alias_in_chat_comands()
+    context.lobby_page.open_alias_menu()
+    context.lobby_page.open_menu()
+    context.lobby_page.switch_to_frame()
```
Comment (timbortnik) [inline features/steps/step_selenium_alias_hipchat.py]: In ideal world, steps should not be aware we are in browser.
Page object is handling anything browser-related.
Test (step) is operating with abstractions that are implementation-agnostic.
How would you name this call if we were using desktop client?

File: pages/lobby_page.py [hunk 15]
```diff
@@ -0,0 +1,139 @@
...
+
+    def identify_frame_status_and_put_data(self):
+        try:
+            self.input_data_in_alias_form()
+            self.input_data_in_alias_name_form()
+            self.open_config()
+            self.input_data_in_alias_form()
+            self.input_data_in_alias_name_form()
+        except:
+            self.open_config()
```
Comment (timbortnik) [inline pages/lobby_page.py]: If just open_config() fails, shall we put self.input_data_in_alias_form() and self.input_data_in_alias_name_form() out of try clause?

File: pages/lobby_page.py [hunk 16]
```diff
@@ -0,0 +1,135 @@
...
+        self.context.driver.find_element_by_css_selector('div>div>.aui-button.aui-button-link').click()
+
+    def put_data_into_the_frame(self):
+            self.input_data_in_alias_form()
+            self.input_data_in_alias_name_form()
+            self.open_config()
+            self.input_data_in_alias_form()
+            self.input_data_in_alias_name_form()
+
+
```
Comment (timbortnik) [inline pages/lobby_page.py]: PEP8?

File: features/steps/step_selenium_alias_hipchat.py [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+    context.lobby_page.random_click()
+    context.lobby_page.open_alias_room()
+    assert context.lobby_page.chat_adding_alias()
+
+@when('we create new alias with click buttons')
+def step_impl(context):
+    context.lobby_page.delete_alias_in_chat_comands()
+    context.lobby_page.open_alias_menu()
+    context.lobby_page.open_menu()
+    context.lobby_page.focus_at_frame()
```
Comment (timbortnik) [inline features/steps/step_selenium_alias_hipchat.py]: Ideally, test steps should not be aware of implementation detail. "Frame" and "iframe" are quite specific to web.

File: pages/lobby_page.py [hunk 18]
```diff
@@ -0,0 +1,135 @@
...
+    def input_data_in_alias_form(self):
+        self.find_alias_form().send_keys("@test")
+
+    def find_alias_form(self):
+        self.context.wait.until(EC.presence_of_element_located((By.NAME, 'alias')))
+        return self.context.driver.find_element_by_name('alias')
+
+    def input_data_in_alias_name_form(self):
+        self.find_form_name().send_keys('HenaYamkoviy')
+        self.context.wait.until(EC.presence_of_element_located((By.XPATH, '//div[text()="HenaYamkoviy"]')))
```
Comment (timbortnik) [inline pages/lobby_page.py:213]: In ideal world, we should use parametric test user name here

**Review by timbortnik — APPROVED** [review]

**Pull request merged**